CoffeeShop.playground/Sources/CoffeeShop.swift

```diff
@@ -0,0 +1,33 @@
+import Foundation
+
+// MARK: - CoffeeShop
+public struct CoffeeShop {
+
+    public let name: String
+    public let location: Location
+
+    public init(name: String, location: Location) {
+        self.name = name
+        self.location = location
+    }
+
+    /// Creates a new instance from the given string by extracting its data delimited by separator
+    ///
+    /// This function throws an error if line does not have the following format:
+    /// <Name>separator<Y Coordinate>separator<X Coordinate>
+    ///
+    /// - Parameter line: The string from where to extract data for the new instance.
+    /// - Parameter separator: The sequence that separates the data.
+    public init?(from line: String, separator: String) {
+        let compoments = line.components(separatedBy: separator)
+        guard !separator.isEmpty,
+            compoments.count == 3,
+            !compoments[0].isEmpty,
+            let x = Double(compoments[2]),
```

Comment:
@eandreea The results are wrong when running the app. This is happening because on read x is the last argument and y is the second one. After comparing with user location you get the results by comparing the user x with local y actually. Please use here x as second arg and y as the third argument.